lib/fizzy/instructions.hpp

```diff
@@ -16,6 +16,9 @@ struct InstructionMetrics
     /// The stack height change caused by the instruction execution,
     /// i.e. stack height _after_ execution - stack height _before_ execution.
     int8_t stack_height_change;
+
+    /// The memory width in bytes the instruction operates on or 0 in case not using the memory.
+    uint8_t memory_width;
```

Comment:
I guess this structure was aligned at 16-bits and we're breaking that now?

Reply:
This is not necessary the best option to add a field to `InstructionMetrics` for sparse values (most are not relevant). This significantly increases the metrics table. But is good enough for now.

It had size 2, alignment 1. Now is size 3 alignment 1.

Reply:
Can go back to the extra switch statement too :wink:

Reply:
I'm for the extra switch statement. For what I checked, it will be implemented as a tiny jump table.

Reply:
Bruh, the instruction metrics route is so much nicer.

Reply:
I benchmarked this and it does not make any difference in parsing so this can stay this way.